### PR TITLE
Z-Wave: add list of used association groups

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

Z-Wave: add list of used association groups
- resolves requirement "RD_AssociationCC_Rev01"

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation on Z-Wave association groups, improving user understanding of direct device communication without a hub.
	- Clarified the role of association groups within Home Assistant, specifically for device management after a factory reset.
	- Corrected grammatical errors and expanded explanations for better clarity.

### CODE REVIEW

File: source/_integrations/zwave_js.markdown [hunk 1]
```diff
@@ -966,3 +966,13 @@ In Home Assistant, it is currently not possible to set the Z-Wave controller int
 ### Including / excluding a controller in an existing network using [classic inclusion](#classic-inclusion-versus-smartstart)
 
 A Z-Wave controller that manages an empty network can also join a different network and act as a secondary controller there. However, with Home Assistant, this is not possible. Home Assistant does not allow the Z-Wave controller to join another network, because Home Assistant acts as the central hub.
+
+## Z-Wave association groups used
+
+An _association_ in Z-Wave terminology is when two or more Z-Wave products communicate directly with one another without the need to communicate via a hub. An _association group_ in Z-Wave terminology is a group of devices that are controlled from a controller device. Z-Wave has a range of association groups that each has different capabilities.
```
Comment (AlCalzone) [inline source/_integrations/zwave_js.markdown]:
```suggestion
An _association_ in Z-Wave terminology is when two or more Z-Wave products communicate directly. This enables devices to communicate with each other without the need to communicate via a hub, or to send unsolicited reports to the central hub.

An _association group_ in Z-Wave terminology is a group of devices that another one will send commands to in certain situations. Association groups and their functionality are specific to the device that sends the commands. Refer to the device manual for details.
```

File: source/_integrations/zwave_js.markdown [hunk 2]
```diff
@@ -966,3 +966,13 @@ In Home Assistant, it is currently not possible to set the Z-Wave controller int
...
+
+In Home Assistant, only one association group is used. It is a group of 1 one device.
```
Comment (AlCalzone) [inline source/_integrations/zwave_js.markdown]:
```suggestion
Home Assistant, a single association group is supported:
```

File: source/_integrations/zwave_js.markdown [hunk 3]
```diff
@@ -966,3 +966,13 @@ In Home Assistant, it is currently not possible to set the Z-Wave controller int
...
+
+- **Group 1**: This is an association group that includes only one device. It is used to send a **Device Reset Locally Notification** after a factory reset.
+
+This association group is used when Home Assistant [resets the Z-Wave controller](#controller) or a Z-Wave end device.
```
Comment (AlCalzone) [inline source/_integrations/zwave_js.markdown]:
```suggestion
This association group is used when Home Assistant [resets the Z-Wave controller](#controller).
```

Comment (AlCalzone) [reply source/_integrations/zwave_js.markdown]: And we should add:
Under normal circumstances, it is not necessary to add a device to this group.

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Comment by AlCalzone** [comment]
Also, does it make sense to move the explanation of what associations are to the "Z-Wave terminology" section?

File: source/_integrations/zwave_js.markdown [hunk 4]
```diff
@@ -972,26 +978,37 @@ In Home Assistant, it is currently not possible to set the Z-Wave controller int
 
 A Z-Wave controller that manages an empty network can also join a different network and act as a secondary controller there. However, with Home Assistant, this is not possible. Home Assistant does not allow the Z-Wave controller to join another network, because Home Assistant acts as the central hub.
 
+## Z-Wave association groups used
+
+Home Assistant, a single [association group](#association-group) is supported:
```
Comment (AlCalzone) [inline source/_integrations/zwave_js.markdown]:
```suggestion
In Home Assistant, a single [association group](#association-group) is supported:
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Review by marcelveldt — APPROVED** [review]

File: source/_integrations/zwave_js.markdown [hunk 5]
```diff
@@ -974,6 +980,16 @@ In Home Assistant, it is currently not possible to set the Z-Wave controller int
 
 A Z-Wave controller that manages an empty network can also join a different network and act as a secondary controller there. However, with Home Assistant, this is not possible. Home Assistant does not allow the Z-Wave controller to join another network, because Home Assistant acts as the central hub.
 
+## Z-Wave association groups used
```
Comment (AlCalzone) [inline source/_integrations/zwave_js.markdown:983]:
```suggestion
## Available Z-Wave association groups
```
maybe? "Used" seems not correct, as we don't use them.
However that could imply they are actually available to users, although they need another controller with a UI to configure them.

Comment (c0ffeeca7) [reply source/_integrations/zwave_js.markdown:983]: <https://github.com/home-assistant/home-assistant.io/pull/34718>

**Pull request merged**